runtime_scan/pkg/scanner/job_managment.go

```diff
@@ -441,6 +441,61 @@ func (s *Scanner) runJob(ctx context.Context, data *scanData) (types.Job, error)
 		return types.Job{}, fmt.Errorf("failed to patch target scan status: %v", err)
 	}
 
+	// since the CLI is not really running, we need to mock it by updating the scan result and mark it as done.
```

Comment:
please remove

Reply:
Actually this is what I did in the first place, but there is a problem with that.
The state is being turned to "attached" after the RunScanningJob is called, so that will cause the waiting for the job to be "Done" to timeout. We need to patch the scanResult to "Done" only after it is moved to "Attached" mode. This is also what the CLI does. 
I agree we need to delete this code from here, but I am not sure how we can mock it. WDYT?